src/include/miopen/generic_search.hpp

```diff
@@ -346,6 +366,9 @@ void CompileAgent(size_t thread_index,
             std::move(current_config), std::move(current_solution), false);
         comp_queue.push(std::move(tup));
     }
+    // Signal to mark thread completion
+    auto tmp = std::make_tuple<PerformanceConfig, ConvSolution, bool>({}, {}, true);
+    comp_queue.push(std::move(tmp));
```

Comment:
This seems like a good catch, I'm surprised generic search wasn't hanging without this.